core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -3418,28 +3415,19 @@ public static RelNode projectMapping(
     return projectFactory.createProject(rel, ImmutableList.of(), exprList, outputNameList);
   }
 
-  /** Predicate for whether a {@link Calc} contains multisets or windowed
-   * aggregates. */
-  public static boolean containsMultisetOrWindowedAgg(Calc calc) {
-    return !(B
-        && RexMultisetUtil.containsMultiset(calc.getProgram())
-        || calc.getProgram().containsAggs());
+  /** Predicate for if a {@link Calc} does not contain windowed aggregates. */
+  public static boolean notContainsWindowedAgg(Calc calc) {
```

Comment:
This look like public API breakage with no clear deprecation plan.

@xndai , @hsyuan was it really intended?

Reply:
hi @vlsi , what is considered as public API, and what is the right process for deprecation? I couldn't find documents on calcite.apache.org

Reply:
So far public API is whatever is `public` and that is not marked as "private/internal" in javadoc.

I've recently added `apiguardian` dependency, so we can use `@API` to classify APIs.

However, in general we consider all Java-public methods to be part of published API.